typespec-extension/src/code-model-builder.ts

```diff
@@ -829,93 +801,101 @@ export class CodeModelBuilder {
 
     // host
     clientContext.hostParameters.forEach((it) => codeModelOperation.addParameter(it));
-    // parameters
-    op.parameters.parameters.map((it) => this.processParameter(codeModelOperation, it, clientContext));
-    // "accept" header
-    this.addAcceptHeaderParameter(codeModelOperation, op.responses);
-    // body
-    if (op.parameters.body) {
-      if (op.parameters.body.property) {
-        if (!isVoidType(op.parameters.body.property.type)) {
-          this.processParameterBody(codeModelOperation, op, op.parameters.body.property);
+    // path/query/header parameters
+    for (const param of httpOperation.parameters) {
+      // if it's paged operation with request body, skip content-type header added by TCGC, as next link call should not have content type header
+      if (
+        (sdkMethod.kind === "paging" || sdkMethod.kind === "lropaging") &&
+        httpOperation.bodyParam &&
+        param.kind === "header"
+      ) {
+        if (param.serializedName.toLocaleLowerCase() === CONTENT_TYPE_KEY) {
+          continue;
         }
-      } else if (op.parameters.body.type) {
-        let bodyType = op.parameters.body.type;
-
-        if (bodyType.kind === "Model") {
-          // try use resource type as round-trip model
-          const resourceType = getResourceOperation(this.program, operation)?.resourceType;
-          if (resourceType && op.responses && op.responses.length > 0) {
-            const resp = op.responses[0];
-            if (resp.responses && resp.responses.length > 0 && resp.responses[0].body) {
-              const responseBody = resp.responses[0].body;
-              const bodyTypeInResponse = this.findResponseBody(responseBody.type);
-              // response body type is resource type, and request body type (if templated) contains resource type
-              if (bodyTypeInResponse === resourceType && isModelReferredInTemplate(bodyType, resourceType)) {
-                bodyType = resourceType;
-              }
-            }
-          }
-
-          this.processParameterBody(codeModelOperation, op, bodyType);
+      }
+      // if the request body is optional, skip content-type header added by TCGC
+      if (httpOperation.bodyParam && httpOperation.bodyParam.optional) {
+        if (param.serializedName.toLocaleLowerCase() === CONTENT_TYPE_KEY) {
+          continue;
         }
```

Comment:
nit, here you remove the header, instead of what you ask in issue "make content-type optional if body is optional".

I am OK you do not change this, but you need to add a `// TODO` to modify this after TCGC decision.

Reply:
issue created: https://github.com/Azure/autorest.java/issues/2930